ip.go

```diff
@@ -57,83 +62,128 @@ type IpCreateResponse struct {
 }
 
 type IpCreateRequest struct {
-	Name         string        `json:"name,omitempty"`
-	Family       int           `json:"family"`
-	LocationUuid string        `json:"location_uuid"`
-	Failover     bool          `json:"failover,omitempty"`
-	ReverseDns   string        `json:"reverse_dns,omitempty"`
-	Labels       []interface{} `json:"labels,omitempty"`
+	Name         string   `json:"name,omitempty"`
+	Family       int      `json:"family"`
+	LocationUuid string   `json:"location_uuid"`
+	Failover     bool     `json:"failover,omitempty"`
+	ReverseDns   string   `json:"reverse_dns,omitempty"`
+	Labels       []string `json:"labels,omitempty"`
 }
 
 type IpUpdateRequest struct {
-	Name       string        `json:"name,omitempty"`
-	Failover   bool          `json:"failover"`
-	ReverseDns string        `json:"reverse_dns,omitempty"`
-	Labels     []interface{} `json:"labels"`
+	Name       string   `json:"name,omitempty"`
+	Failover   bool     `json:"failover"`
+	ReverseDns string   `json:"reverse_dns,omitempty"`
+	Labels     []string `json:"labels"`
+}
+
+type IpEventList struct {
+	List []IpEventProperties `json:"events"`
+}
+
+type IpEvent struct {
+	Properties IpEventProperties `json:"event"`
+}
+
+type IpEventProperties struct {
+	ObjectType    string `json:"object_type"`
+	RequestUuid   string `json:"request_uuid"`
+	ObjectUuid    string `json:"object_uuid"`
+	Activity      string `json:"activity"`
+	RequestType   string `json:"request_type"`
+	RequestStatus string `json:"request_status"`
+	Change        string `json:"change"`
+	Timestamp     string `json:"timestamp"`
+	UserUuid      string `json:"user_uuid"`
 }
 
-func (c *Client) GetIp(id string) (*Ip, error) {
+//GetIp get a specific IP based on given id
+func (c *Client) GetIp(id string) (Ip, error) {
 	r := Request{
-		uri:    apiIpBase + "/" + id,
-		method: "GET",
+		uri:    path.Join(apiIpBase, id),
+		method: http.MethodGet,
 	}
 
-	response := new(Ip)
+	var response Ip
 	err := r.execute(*c, &response)
 
 	return response, err
 }
 
-func (c *Client) GetIpList() (*Ips, error) {
+//GetIpList gets a list of available IPs
+func (c *Client) GetIpList() ([]Ip, error) {
 	r := Request{
 		uri:    apiIpBase,
-		method: "GET",
+		method: http.MethodGet,
 	}
 
-	response := new(Ips)
+	var response Ips
+	var list []Ip
```

Comment:
the `list` is a generic name. Plz, use name reflects the content of the variable e.g., IPs.